packages/jsii-pacmak/lib/targets/go/types/go-type.ts

```diff
@@ -14,9 +14,20 @@ export abstract class GoType {
 
   public constructor(public pkg: Package, public type: Type) {
     this.name = toPascalCase(type.name);
-    // add "_jsiiProxy" postfix to private struct name to avoid keyword
-    // conflicts such as "default". see https://github.com/aws/jsii/issues/2637
-    this.proxyName = `${toCamelCase(type.name)}_jsiiProxy`;
+
+    // Prefix witht he nesting parent name(s), using an _ delimiter.
+    for (
+      let parent = type.nestingParent;
+      parent != null;
+      parent = parent.nestingParent
+    ) {
+      this.name = `${toPascalCase(parent.name)}_${this.name}`;
+    }
+
+    // Add "jsiiProxy_" suffix to private struct name to avoid keyword conflicts
+    // such as "default". See https://github.com/aws/jsii/issues/2637
+    this.proxyName = `jsiiProxy_${this.name}`;
```

Comment:
was this intentionally changed to a prefix. if so, update the comment

Reply:
Yes it was (it looks better for nested type names). And I changed the comment (from `postfix` to `suffix`, because my brain farted).